### PR TITLE
Adding leak features into the model

### PR DESCRIPTION
**Pull request by rewu1993**

### CODE REVIEW

File: include/node.h [hunk 1]
```diff
@@ -93,9 +93,34 @@ class Node {
   //! pdd slope
   double pdd_slope() const { return pdd_slope_; }
 
+  //! leak status
+  bool is_leak() const {
+      return leak_area_ > 0 || false;
```
Comment (kks32) [inline include/node.h:98]: Why do you do an `||`

File: include/node.h [hunk 2]
```diff
@@ -93,9 +93,34 @@ class Node {
...
+  }
+  //! set diameter/area for the leak
+  void leak_diameter(double leak_diameter) {
+    leak_diameter_ = leak_diameter;
+    leak_area_ = std::pow((leak_diameter_ / 2), 2) * pi_;
```
Comment (kks32) [inline include/node.h:103]:
```suggestion
    leak_area_ = pi_ * (0.5 * leak_diameter_) * (0.5 * leak_diameter_);
```

Comment (kks32) [inline include/node.h:103]: Use `M_PI` instead?

File: include/node.h [hunk 3]
```diff
@@ -93,9 +93,34 @@ class Node {
...
+
+  double leak_area() const { return leak_area_; }
+  double leak_discharge() const { return leak_diameter_; }
+  double iter_leak_discharge() { return iter_leak_discharge_; }
+
+  void iter_leak_discharge(double iter_leak_discharge) {
+    iter_leak_discharge_ = iter_leak_discharge;
+  }
+
+  double g() const { return g_; }
```
Comment (kks32) [inline include/node.h:114]: could you rename this to gravity please?

File: include/node.h [hunk 4]
```diff
@@ -124,6 +149,29 @@ class Node {
...
+  //! accleration due to gravity
+  double g_{9.81};
+  //! leak dischage (static and iter)
+  double leak_discharge_{0};
+  double iter_leak_discharge_{0};
+  //! leak coefficients
+  double leak_discharge_coefficient_{0.75};
+
+  //! polynomial coefficients for demand-pressure leak-pressure
+  bool is_pdd_coef1_{false};
```
Comment (kks32) [inline include/node.h:168]: Can we haveit as a vector of coefficients?

File: include/node.h [hunk 5]
```diff
@@ -124,6 +149,29 @@ class Node {
...
+  bool is_pdd_coef2_{false};
+  bool is_leak_coef_{false};
+  Eigen::VectorXd pdd_poly_coef_1_;
```
Comment (kks32) [inline include/node.h:171]: Vector of coefficients?

**Review by kks32 — CHANGES_REQUESTED** [review]

File: include/node.h (hunk 1 above)
Comment (rewu1993) [reply include/node.h:98]: It returns true if leak_area_ > 0 is true false if leak_area_ < 0

File: include/node.h (hunk 2 above)
Comment (rewu1993) [reply include/node.h:103]: What you mean M_PI? From above you you said leak_area_ = pi_ * (0.5 * leak_diameter_) * (0.5 * leak_diameter_); Sorry I am confused

File: include/node.h (hunk 3 above)
Comment (rewu1993) [reply include/node.h:114]: No problem

File: include/node.h (hunk 5 above)
Comment (rewu1993) [reply include/node.h:171]: Oh the bools are not coefficients. They are flags for me to check if coefficients are properly constructed. I guess I can call the construction function inside class constructor instead so I dont need these flags

File: include/node.h (hunk 1 above)
Comment (kks32) [reply include/node.h:98]: `return leak_area_>0` should work or try `return (leak_area_ > 0)? true : false`

Comment (rewu1993) [reply include/node.h:98]: Oh yes! I was dumb. Thanks!

**Pull request merged**